src/components/BigNumberPad.js

```diff
@@ -63,7 +63,7 @@ class BigNumberPad extends React.Component {
                                     style={[styles.flex1, marginLeft]}
                                     text={column === '<' ? column : this.props.toLocaleDigit(column)}
                                     onLongPress={() => this.handleLongPress(column)}
-                                    onPress={() => this.props.numberPressed(column)}
+                                    onPress={() => this.props.numberPressed(column === '<' ? column : this.props.toLocaleDigit(column))}
```

Comment:
How is this change related to this PR? cc: @Santhosh-Sellavel

Reply:
That is how it is supposed to work Internally the values will always be in English for eg. 
`22,23` is really `22.23` which will be saved to the backend. So it was working correctly. 

Manipulating this value here is not a good approach. These changes break the whole concept.

Reply:
@parasharrajat 

We are not saving it as `22,23` or `22.33`. We are rounding it off after multiplying by a hundred, so it's saved as `2233`. I'm not sure about what we broke here?

cc: @sketchydroide

Reply:
by the backend, I mean internally not DB. Why do we have to change this piece of code? It does not look related to the issue.

Reply:
> Why do we have to change this piece of code? It does not look related to the issue. Can anyone please help me understand this?

This keyboard is used only on `IOUAmountPage`, so it seemed logical to make it return the displayed symbol. 
Alternatively we can move this logic to `IOUAmountPage`, change [this](https://github.com/eVoloshchak/App/blob/bcf54e6d4809cf2c0fb7e6bd4fde75214b0186f8/src/pages/iou/steps/IOUAmountPage.js#L143) line to

```javascript
const amount = `${prevState.amount}${this.props.toLocaleDigit(key)}`;
```

Reply:
> Do you think there was any bug on IOUAmountPage that you solved here? What was that? Issue talks about the problem with the rate field on the reimbursement form. 

But the screenshot in the issue description features `Request money` screen. I've asked about this a [couple](https://github.com/Expensify/App/issues/9497#issuecomment-1166304208) of [times](https://github.com/Expensify/App/issues/9497#issuecomment-1167763611), there must have been a miscommunication
> OK, i think i got confused, since the issue description is talking about the Rate input on Reimburce Expenses screen, while screenshot contains IOU Amount Page with an amount input
I'm still not sure which one of the two screens needs to be fixed



> Can we fix the rate field issue without touching IOUAmountPage?

Yes we can
I've checked new.expensify.com and it does indeed support comma for Spanish language on Request Money screen.
I'm guessing we need o revert this PR and clarify what the requirements for this issue are, it it's just for the rate field

Reply:
It seems yes issue details are incorrect. Unfortunately, it was not captured during the proposal review. As a suggestion, please try to get a full consensus before you start if you feel something is off. You can also suggest corrections to the requirements. 

Thanks for understanding. I will open this discussion on the issue so that this PR can be reverted and you can start new.